dora/shell/src/main/java/alluxio/cli/fsadmin/report/JobServiceOutput.java

```diff
@@ -0,0 +1,314 @@
+            mVersion = version;
+        }
+
+        public String getRevision() {
+            return mRevision;
+        }
+
+        public void setRevision(String revision) {
+            mRevision = revision;
+        }
+
+        public int getTaskPoolSize() {
+            return mTaskPoolSize;
+        }
+
+        public void setTaskPoolSize(int taskPoolSize) {
+            mTaskPoolSize = taskPoolSize;
+        }
+
+        public int getUnfinishedTasks() {
+            return mUnfinishedTasks;
+        }
+
+        public void setUnfinishedTasks(int unfinishedTasks) {
+            mUnfinishedTasks = unfinishedTasks;
+        }
+
+        public int getActiveTasks() {
+            return mActiveTasks;
+        }
+
+        public void setActiveTasks(int activeTasks) {
+            mActiveTasks = activeTasks;
+        }
+
+        public List<Double> getLoadAverage() {
+            return mLoadAverage;
+        }
+
+        public void setLoadAverage(List<Double> loadAverage) {
+            mLoadAverage = loadAverage;
+        }
+
+    }
+
+    private static class SerializableStatusSummary {
+        private String mStatus;
+        private long mCount;
+
+        public SerializableStatusSummary(StatusSummary statusSummary){
+            mStatus = statusSummary.getStatus().toString();
+            mCount = statusSummary.getCount();
+        }
+
+        public String getStatus() {
+            return mStatus;
+        }
+
+        public void setStatus(String status) {
+            mStatus = status;
+        }
+
+        public long getCount(){
+            return mCount;
+        }
+
+        public void setCount(long count){
+            mCount = count;
+        }
+    }
+
+    private static class SerializableJobInfo {
+        private String mTimestamp;
+        private long mId;
+        private String mName;
+        private String mStatus;
+        public SerializableJobInfo(JobInfo jobInfo, String dateFormat) {
+            mTimestamp = CommonUtils.convertMsToDate(jobInfo.getLastUpdated(), dateFormat);
+            mId = jobInfo.getId();
+            mName = jobInfo.getName();
+            mStatus = jobInfo.getStatus().toString();
+        }
+
+        public String getTimestamp() {
+            return mTimestamp;
+        }
+
+        public void setTimestamp(String timestamp) {
+            mTimestamp = timestamp;
+        }
+
+        public long getId() {
+            return mId;
+        }
+
+        public void setId(long id) {
+            mId = id;
+        }
+
+        public String getName() {
+            return mName;
+        }
+
+        public void setName(String name) {
+            mName = name;
+        }
+
+        public String getStatus() {
+            return mStatus;
+        }
+
+        public void setStatus(String status) {
+            mStatus = status;
+        }
+    }
+
+    public JobServiceOutput(JobMasterClient jobMasterClient, String dateFormat) throws IOException {
```

Comment:
this class is `Output` so I think it should only be a wrapper of existing information to output, rather than including the logic to poll for the output. So can you call `jobMasterClient.getAllMasterStatus()` outside this class and pass the result to the constructor?

Reply:
Then you might not even need this class? You may just use another `ObjectNode` instead?

Reply:
This part helps to convert given report information to a JSON serializable object. Is it OK to name it "Parser" or something else? There is part of information which is not JSON serializable, so I used an internal Java class to serialize it.